calibrators/marker_radar_lidar_calibrator/README.md

```diff
@@ -0,0 +1,141 @@
+
+## Purpose
+
+The package `marker_radar_lidar_calibrator` allows extrinsic calibration between radar and 3d lidar sensors used in autonomous driving and robotics.
+
+Currently, the calibrator only supports the radar that includes distance and azimuth angle, but without elevation angle. For example, ARS 408 radar can be calibrated with this tool. Also, note that the 3d lidar should have a resolution that is high enough to scan several points on the radar reflector (calibration target).
+
+## Inner-workings / Algorithms
+
+The calibrator is designed to accurately predict the transformation between radar and lidar sensors. It starts by pinpointing the central points of reflectors within lidar pointclouds and radar messages, then aligns these points for precise matching. An SVD-based and a yaw-only rotation estimation algorithm are applied to these correlated points to determine the transformation. Specifically, the calibration process consists of four primary steps: constructing a background model, extracting the foreground to detect reflectors, matching and filtering lidar and radar detections, and finally executing the calibration.
+
+### Step 1: Background model construction
+
+Firstly, given the challenge of reliably detecting reflectors, background models for both lidar and radar are constructed from the lidar pointcloud and radar message within a user-defined calibration area, which lacks any calibration targets (such as radar reflectors). More specifically, these background models consist of uniform binary voxel grids that denote whether each voxel represents the background.
+
+### Step 2: Foreground extraction and reflector detection
+
+After the background models for the lidar and radar are established, we extract the foreground points from incoming lidar pointclouds and radar messages that do not align with the background voxels. All foreground radar points are automatically categorized as potential reflector detections. For foreground lidar points, however, the [reflector](#radar-reflector) detection process is more detailed. We first apply a clustering algorithm to identify clusters, then find the highest point in each cluster, and filter the cluster if the highest point is larger than `reflector_max_height`. Next, we average all points within a reflector_radius from the highest point to estimate the center point of the reflector.
+
+### Step 3: Matching and filtering
+
+Since reflector detections cannot be differentiated directly, we rely on the initial calibration to pair each lidar detection with its closest radar detection, and vice versa. A detection pair is accepted if they are mutually the closest matches. Once a match is made, it is evaluated against existing hypotheses: if it aligns with an existing hypothesis, that hypothesis is updated; if it does not align with any, a new hypothesis is created. When a hypothesis achieves convergence, it is finalized and added to the calibration list.
```

Comment:
"Since reflector detections cannot be differentiated directly"
Is not clear.
 - in the case of multiple targets we do not have a way to distinguish between them
 - as mentioned before, we can not really distinguish between reflector and not reflector detections

Reply:
How to hypotheses converge? I am pretty sure we went through this before

Reply:
fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/a326bb73cc3a700e6908bc53e13094da0a813ef3

Reply:
a Kalman filter
or
Kalman filters

Reply:
Almost a nit but:
A detection pair is accepted if they are mutually `their` closest matches

Reply:
fxied in https://github.com/tier4/CalibrationTools/pull/164/commits/a5bc52745c18ea1d5cee145f245de2e67972ea8a